### PR TITLE
Quality Deliveries table: drop duplicates on insert

### PR DESCRIPTION
**Pull request by mathisbrrt**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205274506786268

### CODE REVIEW

File: lox_services/persistence/database/remove_duplicates.py [hunk 1]
```diff
@@ -114,6 +115,70 @@ def remove_duplicate_refunds(dataframe: pd.DataFrame) -> pd.DataFrame:
...
+    are found, they are removed from the DataFrame. The function also ensures that the
+    datetime column is formatted correctly for comparison.
+
+    Args:
+        dataframe (pd.DataFrame): The input DataFrame containing delivery data.
+
+    Returns:
+        pd.DataFrame: A DataFrame with duplicate deliveries removed, if any.
+    """
+    print("Checks for duplicate rows in db..")
```
Comment (natasazekic-loxsolution) [inline lox_services/persistence/database/remove_duplicates.py:133]: NIT: I would put print after empty check.

File: lox_services/persistence/database/remove_duplicates.py [hunk 2]
```diff
@@ -114,6 +115,70 @@ def remove_duplicate_refunds(dataframe: pd.DataFrame) -> pd.DataFrame:
...
+        dataframe["tracking_number"].astype(str)
+        + dataframe["status"].astype(str)
+        + dataframe["formated_datetime"].astype(str)
+    )
+
+    sql_query = """
+        SELECT
+            tracking_number || status || date_time as concat_values
+        FROM
+            `developmentproject-269810.InvoicesData.Deliveries`
```
Comment (natasazekic-loxsolution) [inline lox_services/persistence/database/remove_duplicates.py:154]: Definitely not the dev project hehe

File: lox_services/persistence/database/remove_duplicates.py [hunk 3]
```diff
@@ -114,6 +115,70 @@ def remove_duplicate_refunds(dataframe: pd.DataFrame) -> pd.DataFrame:
...
+
+    if deliveries_already_in_db.empty:
+        print("Nothing to delete")
+        return dataframe.drop(columns=["concat_values", "formated_datetime"])
+
+    else:
+        original_length = len(dataframe.index)
+        duplicate_values = deliveries_already_in_db["concat_values"].unique().tolist()
+        dataframe = dataframe.loc[~dataframe["concat_values"].isin(duplicate_values)]
+        print(original_length - len(dataframe.index), "Rows removed.")
```
Comment (natasazekic-loxsolution) [inline lox_services/persistence/database/remove_duplicates.py:178]: Why not `print(f"Number of rows removed: ", original_length - len(dataframe.index))`

**Review by natasazekic-loxsolution — APPROVED** [review]

File: lox_services/persistence/database/remove_duplicates.py (hunk 1 above)
Comment (mathisbrrt) [reply lox_services/persistence/database/remove_duplicates.py:133]: There is already one print in the insert function

File: lox_services/persistence/database/remove_duplicates.py [hunk 4]
```diff
@@ -114,6 +115,70 @@ def remove_duplicate_refunds(dataframe: pd.DataFrame) -> pd.DataFrame:
...
+            (
+                "concat_values",
+                BQParameterType.STRING,
+                dataframe["concat_values"].unique().tolist(),
+            )
+        ],
+    )
+
+    if deliveries_already_in_db.empty:
+        print("Nothing to delete")
```
[the remark this answers is not in the document]
Comment (mathisbrrt) [reply lox_services/persistence/database/remove_duplicates.py:171]: We have one in all other functions

File: lox_services/persistence/database/remove_duplicates.py (hunk 1 above)
Comment (mathisbrrt) [reply lox_services/persistence/database/remove_duplicates.py:133]: Ok I got your point now

**Pull request merged**